providers/oracle/src/airflow/providers/oracle/hooks/oracle.py

```diff
@@ -346,6 +365,22 @@ def bulk_insert_rows(
         """
         if not rows:
             raise ValueError("parameter rows could not be None or empty iterable")
+
+        if not _is_valid_identifier(table):
+            self.log.error("Invalid table name detected: %s", table)
+            raise ValueError(f"Invalid table name: {table}")
+        if target_fields:
+            for field in target_fields:
+                if not _is_valid_identifier(field):
+                    self.log.error("Invalid column name detected: %s", field)
+                    raise ValueError(f"Invalid column name: {field}")
+        if sequence_column and not _is_valid_identifier(sequence_column):
+            self.log.error("Invalid sequence column name detected: %s", sequence_column)
+            raise ValueError(f"Invalid sequence column name: {sequence_column}")
+        if sequence_name and not _is_valid_identifier(sequence_name):
+            self.log.error("Invalid sequence name detected: %s", sequence_name)
+            raise ValueError(f"Invalid sequence name: {sequence_name}")
```

Comment:
Maybe instead of repeating logic you could extract common logic to an internal function and use it in both `insert_rows` and `build_insert_rows`?